lib/CL/devices/basic/basic.c

```diff
@@ -543,7 +548,10 @@ pocl_basic_run
   pc->local_size[0] = cmd->command.run.local_x;
   pc->local_size[1] = cmd->command.run.local_y;
   pc->local_size[2] = cmd->command.run.local_z;
-  
+
+  pocl_set_drm ();
```

Comment:
You need to save and restore the previous state of drm and ftz flags. Otherwise, if user of pocl has set them on already, it produces unexpected behavior for him.